exopy_hqc_legacy/tasks/tasks/instr/pna_tasks.py

```diff
@@ -350,6 +350,11 @@ def perform(self):
                     self.channel_driver.prepare_measure(meas_name, self.window,
                                                         i+1, clear)
                     clear = False
+                    
+        # changing format so that it does not ouput ascii which is default
+        # for PNA Agilent N5222A and messes up with the SaveFile(h5) tasks
```

Comment:
I am still curious of what is the issue.

Reply:
I finally succeeded in resetting the output format to ASCII (which corresponds to 'ASC,+0' and not 'ASCii,0' like in the PNA documentation), and then when I want to save data, the consol prints:

`Logger parametrised
Process running
Task built
Check successful
ERROR: The following unhandled exception occured :
Traceback (most recent call last):
File "C:\Users\Marius\anaconda3\envs\exopy\lib\site-packages\numpy\lib\npyio.py", line 1447, in savetxt
v = format % tuple(row) + newline
TypeError: must be real number, not numpy.str_

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
File "c:\users\marius\documents\github\exopy\exopy\tasks\tasks\base_tasks.py", line 1148, in perform
child.perform_()
File "c:\users\marius\documents\github\exopy\exopy\tasks\tasks\decorators.py", line 86, in decorator
return function_to_decorate(*args, **kwargs)
File "c:\users\marius\documents\github\exopy\exopy\tasks\tasks\decorators.py", line 307, in wrapper
return perform(obj, *args, **kwargs)
File "c:\users\marius\documents\github\exopy_hqc_legacy\exopy_hqc_legacy\tasks\tasks\util\save_tasks.py", line 430, in perform
numpy.savetxt(self.file_object, array_to_save, delimiter='\t')
File "", line 6, in savetxt
File "C:\Users\Marius\anaconda3\envs\exopy\lib\site-packages\numpy\lib\npyio.py", line 1451, in savetxt
% (str(X.dtype), format))
TypeError: Mismatch between array dtype ('(numpy.record, [('f0', 'Process shuting down
`

This error disappears if I set the output format like in the section of the code that you commented

Reply:
That looks mostly like a bug in how you read the data. As I mentioned in my review I could not find the definition for ascii and most likely parsing the ascii fails somehow (I am surprised it does so silently however) and you end up with a numpy array containing strings.

Reply:
I did not dare to add something to visa_tools.py but now I am going to do it